lib/iris/cube.py

```diff
@@ -3131,6 +3084,10 @@ def __eq__(self, other):
         if isinstance(other, Cube):
             result = self.metadata == other.metadata
 
+            # check the cube fill-value.
+            if result:
+                result = self.fill_value == other.fill_value
```

Comment:
> so is this a general issue that should be raised beyond this PR?

Yeah, it probably is (we should probably make it very clear somewhere what factors determine cube equality!).

I sort of hate this:

```python
>>> np.array(1) == np.array(1.0)
True
```